contracts/Market.sol

```diff
@@ -598,21 +606,19 @@ contract Market is PausableAccessPreset {
    * @param amounts An array of amounts being allocated from each corresponding removal token.
    * @param suppliers An array of suppliers
    *
-   * todo permission `fulfillOrder` now that it is external (or figure out how to use calldata with internal fns)
    * todo use correct check-effects pattern in `fulfillOrder`
    */
-  function fulfillOrder(
+  function _fulfillOrder(
     uint256 certificateAmount,
     address operator,
     address recipient,
     uint256 numberOfRemovals,
-    uint256[] calldata ids,
-    uint256[] calldata amounts,
+    uint256[] memory ids,
+    uint256[] memory amounts,
     address[] memory suppliers
-  ) external {
-    // todo verify changes to `fulfillOrder` (memory->calldata arr args) that enabled [:index] arr slicing syntax is ok
-    uint256[] memory batchedIds = ids[:numberOfRemovals];
-    uint256[] memory batchedAmounts = amounts[:numberOfRemovals];
+  ) internal {
```

Comment:
based on the answer to [this post](https://ethereum.stackexchange.com/questions/123169/can-calldata-be-used-in-every-function-visibility), I believe that `calldata` arguments really do have to come from the data that was submitted with the initial transaction externally, because it lives in a different place in memory.  I've had unexpected behavior when marking params with `calldata` that are actually called internally with non-calldata arguments.  Jaycen was trying to obtain gas savings on this function using calldata but it's not appropriate here